script.js

```diff
@@ -1,4 +1,523 @@
+var playerBetsTurn = 1; // Default start with player 1
+var splitQnAsked = false; // Tracks if split question has been answered
+var splitTurn = 1; // Default to start split turn on 1
+var splitMode = false; // To initiate split or not
+var newDeck = shuffleCards(makeDeck());
+var master = {};
+
+var reset = function () {
+  newDeck = shuffleCards(makeDeck());
+  for (var i = 0; i < numOfPlayers; i += 1) {
+    var currentRef = i + 1;
+    if (currentRef == numOfPlayers) {
+      master["dealer"] = { decks: [], blackjackValues: [] };
+    } else {
+      master[currentRef].decks = [];
+      master[currentRef].blackjackValues = [];
+      master[currentRef].playerBets = 0;
+      master[currentRef].split = false;
+    }
+  }
+  playerTurn = 1;
+  gameMode = "bets";
+  playerBetsTurn = 1;
+};
+
+// ===== MAIN FUNCTION ===== //
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var dealer = numOfPlayers; // Dealer is always the last to play
+
+  //PHASE 1: PICK NUMBER OF PLAYERS
+  if (gameMode == "pick number of players") {
+    if (input >= 2) {
+      numOfPlayers = Number(input);
+      myOutputValue += `You have selected <b>${numOfPlayers} players</b> to play. Press '<b>Submit</b>' to continue...`;
+      gameMode = "bets";
+      createPlayers(master, numOfPlayers);
+    } else {
+      myOutputValue +=
+        `Please <b>input the number of players</b> in the game (at least 2, inclusive of dealer).<br>` +
+        playersImage;
+    }
+  }
+  //PHASE 2: BETS
+  else if (gameMode == "bets") {
+    myOutputValue += `Player ${playerBetsTurn}, you currently have <b>${master[playerBetsTurn].playerPoints} points</b>.<br><br>How many points (at least 1) would you like to wager?${wagerImage}`;
+    // Check if any players still have points
+    if (sumOfAllPoints(master) == 0) {
+      myOutputValue = `All players do not have any points left... <b>GAME OVER</b>!${gameOverImage}<br><br>Please <b>refresh</b> the page to restart the game.`;
+    }
+    // To skip current player's bet if they have no more points left
+    else if (master[playerBetsTurn].playerPoints == 0) {
+      myOutputValue = `Player ${playerBetsTurn}, you do not have any points left... Press '<b>Submit</b> to continue...`;
+      master[playerBetsTurn].playerBets = 0;
+      if (playerBetsTurn == numOfPlayers - 1) {
+        gameMode = "draw cards";
+      } else {
+        playerBetsTurn += 1;
+      }
+    } else if (input > 0 && input <= master[playerBetsTurn].playerPoints) {
+      master[playerBetsTurn].playerBets = Number(input);
+      myOutputValue = `Player ${playerBetsTurn}, you have chosen to wager <b>${master[playerBetsTurn].playerBets} points</b>.<br><br> Press '<b>Submit</b>' to continue...${postWagerImage}`;
+      if (playerBetsTurn == numOfPlayers - 1) {
+        gameMode = "draw cards";
```

Comment:
neat but not very clear
if dealer Index is declared before, it can be clearer
dealerTurnNum = numOfPlayer-1

if( playerBetTurn == dealerTurnNum)